### PR TITLE
impl_web_hook_action_listener

### PR DESCRIPTION
**Pull request by Thumimku**
## Purpose

- Refactor sql files (adding spaces).
- Implement Web hook action listener, builder, component.
- Implement Action Listener Impl class to handle create manage deploy revisions (Decouple iteration from deployment)

### CODE REVIEW

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/webhook/WebHookActionListener.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public WebHookActionListener(RepositoryManager repo, ConfigDeployer configDeployer,
+                                 String remoteFetchConfigurationId, int tenantId) {
+
+        super(repo, configDeployer, remoteFetchConfigurationId, tenantId);
+    }
+
+    @Override
+    public void iteration() {
+
+        super.iteration();
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/webhook/WebHookActionListener.java:43]: We would not need to reimplement this method as it only calls super method

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/ActionListenerImpl.java [hunk 2]
```diff
@@ -46,34 +45,43 @@
...
-        this.frequency = frequency;
         this.remoteFetchConfigurationId = remoteFetchConfigurationId;
-        this.deploymentRevisionDAO = new DeploymentRevisionDAOImpl();
         this.tenantId = tenantId;
+        this.deploymentRevisionDAO = new DeploymentRevisionDAOImpl();
         this.seedRevisions();
     }
 
+    @Override
+    public void iteration() {
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/ActionListenerImpl.java:75]: Looks like this method is not an iteration at all. This is better rem=named as "execute" or "process"

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/webhook/WebHookActionListener.java (hunk 1 above)
Comment (ruwanta) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/webhook/WebHookActionListener.java:43]: Is this method run in a scheduled task ?

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/ActionListenerImpl.java [hunk 3]
```diff
@@ -46,34 +45,43 @@
...
-        this.frequency = frequency;
         this.remoteFetchConfigurationId = remoteFetchConfigurationId;
-        this.deploymentRevisionDAO = new DeploymentRevisionDAOImpl();
         this.tenantId = tenantId;
+        this.deploymentRevisionDAO = new DeploymentRevisionDAOImpl();
         this.seedRevisions();
     }
 
+    @Override
+    public void execute() {
```
Comment (brionmario) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/ActionListenerImpl.java:75]: Add a new line space after method name.

Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/ActionListenerImpl.java:75]: Thank you, This comment and your previous comments were fixed in
https://github.com/wso2-extensions/identity-fetch-remote/pull/40
https://github.com/wso2-extensions/identity-fetch-remote/pull/41
https://github.com/wso2-extensions/identity-fetch-remote/pull/42
https://github.com/wso2-extensions/identity-fetch-remote/pull/43
https://github.com/wso2-extensions/identity-fetch-remote/pull/44
https://github.com/wso2-extensions/identity-fetch-remote/pull/45.
Hence I resolve this conversation.

**Pull request merged**